src/core/border_router/routing_manager.cpp

```diff
@@ -231,6 +283,13 @@ void RoutingManager::Stop(void)
         DeprecateOnLinkPrefix();
     }
 
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_NAT64_ENABLE
+    if (mIsAdvertisingLocalNat64Prefix)
+    {
+        RemoveExternalRoute(mLocalNat64Prefix);
+        mIsAdvertisingLocalNat64Prefix = false;
```

Comment:
`mIsAdvertisingLocalNat64Prefix` seems unnecessary?
`RemoveExternalRoute` does nothing when the prefix was not added.

Reply:
Prefer not to call RemoveHasRoutePrefix when local prefix is not advertised.
RemoveHasRoutePrefix will return kErrorNotFound and log it with "Failed to remove external route xxx". 
This can be confused with the scenario that a local prefix is expected to be advertised but is failed to be found and removed.

Reply:
sgtm.